server/server.go

```diff
@@ -2,44 +2,47 @@ package server
 
 import (
 	"context"
-	"log"
 	"net/http"
 	"os"
 	"os/signal"
 	"time"
 
+	"github.com/gin-contrib/zap"
+	"go.uber.org/zap"
+
 	"github.com/gin-gonic/gin"
 )
 
-const shutdownTimeout = 5 * time.Second
+const shutdownTimeout = 1 * time.Second
 
 // Run listens given address.
-func Run(listenAddr string) {
+func Run(listenAddr string, logger *zap.Logger) {
 	server := http.Server{
 		Addr:    listenAddr,
-		Handler: setupRouter(),
+		Handler: setupRouter(logger),
 	}
 
 	go func() {
 		if err := server.ListenAndServe(); err != http.ErrServerClosed {
-			log.Fatal(err) //TODO: add logging
+			logger.Fatal("server stopped", zap.Error(err))
 		}
 	}()
 
 	quit := make(chan os.Signal)
 	signal.Notify(quit, os.Interrupt)
 	<-quit
-	log.Println("Shutdown Server ...") //TODO: add logging
+	logger.Info("shutdown server")
```

Comment:
could you please show how logs will look like?

Reply:
In json mode (in production environment):
```
{"level":"info","ts":1552381495.3769605,"caller":"server/server.go:34","msg":"shutdown server"}
```

In text mode (in development environment):
```
2019-03-12T12:06:34.156+0300	INFO	server/server.go:34	shutdown server
```